File: articles/employee-motivation-organizations/fortran/motivation_system_score.f90
! Employee Motivation in Organizations
! Simple Fortran scoring demonstration.
!
! Responsible-use scope:
! Synthetic-data research, methods demonstration, institutional learning,
! and reproducible workflows only. Not for employment, productivity ranking,
! motivation scoring, loyalty scoring, or individual decisions.

program motivation_system_score_program
  implicit none
  real :: autonomy_support, competence_development, recognition_quality
  real :: institutional_trust, goal_clarity, fairness_perception
  real :: workload_strain, control_pressure, relatedness
  real :: meaningful_work, psychological_safety, leadership_credibility
  real :: score

  autonomy_support = 84.0
  competence_development = 80.0
  recognition_quality = 79.0
  institutional_trust = 82.0
  goal_clarity = 83.0
  fairness_perception = 81.0
  workload_strain = 22.0
  control_pressure = 18.0
  relatedness = 82.0
  meaningful_work = 84.0
  psychological_safety = 80.0
  leadership_credibility = 82.0

  score = 0.12 * autonomy_support + 0.11 * competence_development + &
          0.10 * recognition_quality + 0.12 * institutional_trust + &
          0.10 * goal_clarity + 0.11 * fairness_perception + &
          0.10 * relatedness + 0.11 * meaningful_work + &
          0.10 * psychological_safety + 0.10 * leadership_credibility - &
          0.08 * workload_strain - 0.07 * control_pressure

  print *, "Synthetic motivation quality score:", score
  print *, "Responsible-use reminder: institutional learning only."
end program motivation_system_score_program

File: articles/employee-motivation-organizations/go/motivation_system_score.go
// Employee Motivation in Organizations
// Simple Go scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, institutional learning,
// and reproducible workflows only. Not for employment, productivity ranking,
// motivation scoring, loyalty scoring, or individual decisions.

package main

import "fmt"

type MotivationSystemProfile struct {
	AutonomySupport        float64
	CompetenceDevelopment float64
	RecognitionQuality    float64
	InstitutionalTrust    float64
	GoalClarity           float64
	FairnessPerception    float64
	WorkloadStrain        float64
	ControlPressure       float64
	Relatedness           float64
	MeaningfulWork        float64
	PsychologicalSafety   float64
	LeadershipCredibility float64
}

func MotivationQualityScore(p MotivationSystemProfile) float64 {
	return 0.12*p.AutonomySupport +
		0.11*p.CompetenceDevelopment +
		0.10*p.RecognitionQuality +
		0.12*p.InstitutionalTrust +
		0.10*p.GoalClarity +
		0.11*p.FairnessPerception +
		0.10*p.Relatedness +
		0.11*p.MeaningfulWork +
		0.10*p.PsychologicalSafety +
		0.10*p.LeadershipCredibility -
		0.08*p.WorkloadStrain -
		0.07*p.ControlPressure
}

func main() {
	profile := MotivationSystemProfile{84, 80, 79, 82, 83, 81, 22, 18, 82, 84, 80, 82}
	fmt.Printf("Synthetic motivation quality score: %.2f\n", MotivationQualityScore(profile))
	fmt.Println("Responsible-use reminder: institutional learning only.")
}
